File: SWEA/D3/1209. ［S／W 문제해결 기본］ 2일차 － Sum/［S／W 문제해결 기본］ 2일차 － Sum.py
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class Solution {
  public static void main(String[] args) throws Exception {

    /**
     * 0. 입력파일 읽어들이기
     */
    BufferedReader in = new BufferedReader(new InputStreamReader(System.in));

    // 결과를 한 번에 출력하기 위한 StringBuilder
    StringBuilder sb = new StringBuilder();

    for (int test_case = 1; test_case <= 10; test_case++) {
      sb.append("#" + test_case + " ");

      int T = Integer.parseInt(in.readLine());
      // 여러분의 알고리즘 코드 작성하기

      /*
       * 1. 입력 파일 객체화
       */

      int[][] arr = new int[100][100];
      for (int i = 0; i < 100; i++) {
        String[] inputs = in.readLine().split(" ");
        for (int j = 0; j < 100; j++) {
          arr[i][j] = Integer.parseInt(inputs[j]);
        }
      }

      /*
       * 2. 알고리즘 풀기
       */

      int result = 0;

      // 가로
      for (int y = 0; y < 100; y++) {
        int sumRow = 0;
        for (int x = 0; x < 100; x++) {
          sumRow += arr[y][x];
        }
        result = Math.max(sumRow, result);
      }

      // 세로
      for (int x = 0; x < 100; x++) {
        int sumRow = 0;
        for (int y = 0; y < 100; y++) {
          sumRow += arr[y][x];
        }
        result = Math.max(sumRow, result);
      }

      // 대각선 좌상 우하
      int sumRow = 0;
      for (int y = 0; y < 100; y++) {
        sumRow += arr[y][y];
      }
      result = Math.max(sumRow, result);

      // 대각선 우상 좌하
      sumRow = 0;
      for (int y = 0; y < 100; y++) {
        sumRow += arr[y][99 - y];
      }

      result = Math.max(sumRow, result);


      /*
       * 정답 출력
       */
      sb.append(result).append("\n");
    }

    System.out.println(sb);
  }
}

File: SWEA/D3/1209. ［S／W 문제해결 기본］ 2일차 － Sum/［S／W 문제해결 기본］ 2일차 － Sum.java
import java.util.*;
import java.io.*;

public class Solution {
	private static final int SIZE = 100;
	private static int[][] map = new int[SIZE][SIZE];

	public static void main(String[] args) throws Exception {

		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

		int T = 10;
		StringBuilder sb = new StringBuilder();

		for (int test_case = 1; test_case <= T; test_case++) {
			sb.append("#").append(test_case).append(" ");

			int N = Integer.parseInt(br.readLine());

			for (int i = 0; i < SIZE; i++) {
				StringTokenizer st = new StringTokenizer(br.readLine());
				for (int j = 0; j < SIZE; j++) {
					map[i][j] = Integer.parseInt(st.nextToken());
				}
			}

			sb.append(findMaxSum()).append("\n");
		}

		System.out.println(sb);
	}

	private static int findMaxSum() {
		int maxSum = 0;
		int diagonalSum = 0, antiDiagonalSum = 0;

		for (int i = 0; i < SIZE; i++) {
			int rowSum = 0, colSum = 0;
			for (int j = 0; j < SIZE; j++) {
				rowSum += map[i][j];
				colSum += map[j][i];
			}
			diagonalSum += map[i][i];
			antiDiagonalSum += map[i][SIZE - i - 1];
			maxSum = Math.max(maxSum, Math.max(rowSum, colSum));
		}
		return Math.max(maxSum, Math.max(diagonalSum, antiDiagonalSum));
	}
}